provider/resource_user.go

```diff
@@ -0,0 +1,114 @@
+package provider
+
+import (
+	"encoding/json"
+
+	"github.com/BESTSELLER/terraform-provider-harbor/client"
+	"github.com/BESTSELLER/terraform-provider-harbor/models"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+)
+
+func resourceUser() *schema.Resource {
+	return &schema.Resource{
+		Schema: map[string]*schema.Schema{
+			"username": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"password": {
+				Type:      schema.TypeString,
+				Required:  true,
+				Sensitive: true,
+			},
+			"full_name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"email": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"admin": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  false,
+			},
+			"comment": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+		},
+		Create: resourceUserCreate,
+		Read:   resourceUserRead,
+		Update: resourceUserUpdate,
+		Delete: resourceUserDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+	}
+}
+
+func resourceUserCreate(d *schema.ResourceData, m interface{}) error {
+	apiClient := m.(*client.Client)
+
+	body := client.UserBody(d)
+
+	resp, err := apiClient.SendRequest("POST", models.PathUsers, &body, 201)
+	if err != nil {
+		return err
+	}
+
+	var jsonData models.UserBody
+	json.Unmarshal([]byte(resp), &jsonData)
+
+	d.SetId(jsonData.Localation)
+	return resourceUserRead(d, m)
+}
+
+func resourceUserRead(d *schema.ResourceData, m interface{}) error {
+	apiClient := m.(*client.Client)
+	resp, err := apiClient.SendRequest("GET", d.Id(), nil, 200)
+	if err != nil {
+		return err
+	}
+	var jsonData models.UserBody
+	json.Unmarshal([]byte(resp), &jsonData)
+
+	d.Set("username", jsonData.Username)
+	d.Set("full_name", jsonData.Realname)
+	d.Set("email", jsonData.Email)
+	d.Set("admin", jsonData.SysadminFlag)
+	d.Set("comment", jsonData.Comment)
+
+	return nil
+}
+
+func resourceUserUpdate(d *schema.ResourceData, m interface{}) error {
+	apiClient := m.(*client.Client)
+
+	body := client.UserBody(d)
+	_, err := apiClient.SendRequest("PUT", d.Id(), body, 200)
+	if err != nil {
+		return err
+	}
+
+	_, err2 := apiClient.SendRequest("PUT", d.Id()+"/sysadmin", body, 200)
```

Comment:
No need to create a  new variable each time, you can just re-use the same `err`
So it would just be
```golang
_, err = apiClient.SendRequest("PUT", d.Id()+"/sysadmin", body, 200)
```

Reply:
fixed